docs/pages/machine-id/guides/jenkins.mdx

```diff
@@ -0,0 +1,149 @@
+---
+title: Jenkins
+description: Using Teleport Machine ID with Jenkins
+---
+
+Jenkins is a open source automation server that is frequently used to build
+Continuous Integration and Continuous Delivery (CI/CD) pipelines.
+
+In this guide, we will demonstrate how to migrate Jenkins workloads to
+Machine ID with minimal changes to your pipelines.
+
+## Prerequisites
+
+You will need the following tools to use Teleport with Jenkins.
+
+- [Teleport Enterprise](../../enterprise/introduction.mdx) or [Teleport Cloud](../../cloud/introduction.mdx) >= 9.0.0 
+- `ssh` OpenSSH tool
+- `Jenkins`
+
+In addition, if you already have not done so, follow the
+[Machine ID Getting Started Guide](../../docs/pages/machine-id/getting-started.mdx)
+to create a bot user and start Machine ID.
+
+## Architecture
+
+Before we begin, it should be noted that Jenkins is a tool that is notoriously
+difficult to secure. Machine ID is one part of securing your infrastructure,
+but it alone is not sufficient. Below we will provide some basic guidance which
+can help improve the security posture of your Jenkins install.
+
+The simplest Jenkins deployments have the controller and agents run on the
+same host. This deployment model is simple to get started with, however any
+compromise of the `jenkins` user within a single pipeline can lead to the
+compromise of your entire CI/CD infrastructure.
+
+A slightly more complex, but more secure, deployment is running your Jenkins
+controllers and agents on different hosts and pinning workloads to specific
+agents. This is a improvement over the simple deployment because you can limit
+the blast radius of the compromise of a single pipeline to a subset of your
+CI/CD infrastructure instead of all of your infrastructure.
+
+We strongly encourage the use of the second deployment model whenever possible,
+with ephemeral hosts when possible. When using Machine ID with this model,
```

Comment:
We might want to qualify "ephemeral hosts" a bit since we only really support them now via IAM auth and still don't have a good story for ephemeral on-prem use.